canonical/library.canonical.com
# Add function for positioning all items in doc hierarchy based on leading number

## Done

- This PR allows ordering of side navigation items. Previously ordering was only applied to top level items.
- Create a new function `insert_based_on_position` that when building the `doc_hierarchy`, inserts each document based on a 'position' value. The value is defined in google docs by prepending a leading number (ex. '10-Organisation'). This number then becomes the position value ('10') and is subsequently removed from the name.
- Updated some comments on the code
- Small refactor to use this function

## QA

- Open the [top level of google drive](https://drive.google.com/drive/u/0/folders/1QLSNL1QhMMHJmDVFyTXoQ2V6RBtc8mjx)
- Go to https://library-canonical-com-68.demos.haus/
- See that the order of documents on google drive is the same as on the library
- Open [a nested folder in google drive](https://drive.google.com/drive/u/0/folders/1OFK1M74Q7gLIwzn3B_sAeidv4UQhSKMc)
- Go to https://library-canonical-com-68.demos.haus/tests-and-issues
- See that the order of documents on google drive is the same as on the library
- From within the [test folder](https://drive.google.com/drive/u/0/folders/1OFK1M74Q7gLIwzn3B_sAeidv4UQhSKMc), make some changes to the names and see the changes are reflected in the library.


## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-3827

## Screenshots

![image](https://github.com/canonical/library.canonical.com/assets/58276363/1ee4307e-6cd7-4c7f-a92b-9847ab46ce88)
![image](https://github.com/canonical/library.canonical.com/assets/58276363/e3539804-4e48-4758-8259-c9fef90c98d3)
